x-pack/plugins/apm/public/components/app/TransactionDetails/Distribution/index.tsx

```diff
@@ -52,11 +53,31 @@ interface Props {
 
 export class Distribution extends Component<Props> {
   public formatYShort = (t: number) => {
-    return `${t} ${unitShort(this.props.urlParams.transactionType)}`;
+    return i18n.translate(
+      'xpack.apm.transactionDetails.transactionsDurationDistributionChart.unitShortLabel',
+      {
+        defaultMessage:
+          '{transCount} {transType, select, request {req.} other {trans.}}',
+        values: {
+          transCount: t,
+          transType: this.props.urlParams.transactionType as string
```

Comment:
Why `as string`? If it's `undefined` it will default to `other` (which is "trans."), right?

Reply:
Correct, it will chose `trans.`.
I added `as string` to fix broken TS. `this.props.urlParams.transactionType` can be `undefined`, but value ( in [key: value] pair) is required to be `string`, `number` or `Date`.
I'll create a PR to expand valid types for value.

Reply:
I created a PR #28907 to extend types for this case. After it is merged, I will update this PR.